### PR TITLE
Switch from jenkins to github actions

### PR DESCRIPTION
**Pull request by joakimen**

### CODE REVIEW

**Comment by joakimen** [comment]
Used a cache configuration inspired by [this article](https://adamj.eu/tech/2023/11/02/github-actions-faster-python-virtual-environments/), which is a python-adaption of the [actions/cache readme](https://github.com/actions/cache?tab=readme-ov-file#creating-a-cache-key).

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      - uses: capralifecycle/actions-lib/check-runtime-dependencies@42cbe330ccc0282f04edbf0a6ee8928b5b2c4df0 # v1.1.0
+
+      - uses: actions/setup-node@b39b52d1213e96004bfcb1c61a8a6fa8ab84f3e8 # v4.0.1
+        with:
+          node-version: 18.19.0
+
+      - uses: capralifecycle/actions-lib/configure-npm@42cbe330ccc0282f04edbf0a6ee8928b5b2c4df0 # v1.1.0
+
+      - uses: actions/setup-python@v4
+        id: setup_python
```
Comment (stekern) [inline .github/workflows/ci.yml:36]: nit: kebab-case
```suggestion
        id: setup-python
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+      - uses: capralifecycle/actions-lib/check-runtime-dependencies@42cbe330ccc0282f04edbf0a6ee8928b5b2c4df0 # v1.1.0
+
+      - uses: actions/setup-node@b39b52d1213e96004bfcb1c61a8a6fa8ab84f3e8 # v4.0.1
+        with:
+          node-version: 18.19.0
+
+      - uses: capralifecycle/actions-lib/configure-npm@42cbe330ccc0282f04edbf0a6ee8928b5b2c4df0 # v1.1.0
+
+      - uses: actions/setup-python@v4
```
Comment (stekern) [inline .github/workflows/ci.yml:35]:
```suggestion
      - uses: actions/setup-python@v5
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        id: setup_python
+        with:
+          python-version: 3.8.18
+
+      - name: python - setup dependency cache
+        uses: actions/cache@v3
```
Comment (stekern) [inline .github/workflows/ci.yml:41]:
```suggestion
        uses: actions/cache@v4
```

**Review by stekern — APPROVED** [review]
Looks good besides some minor comments 🚀 

(nit: I prefer using `fix(ci)` for this to avoid unnecessarily changing the semantic version more than we need...)

**Pull request merged**